src/main/java/net/minecraftforge/common/config/ConfigManager.java

```diff
@@ -202,17 +230,52 @@ private static void createConfig(String modid, String category, Configuration cf
         {
             if (category.isEmpty())
                 throw new RuntimeException("Can not specify a primitive field when the category is empty: " + f.getDeclaringClass() +"/" + f.getName());
-            prop = adapter.getProp(cfg, category, f, instance, comment);
-            set(instance, f, adapter.getValue(prop));
+            if(!loading)
+            {
+                Property property = cfg.getCategory(category).get(getName(f));
+                Object fieldValue = get(instance, f);
+                Object propValue = adapter.getValue(property);
+                if(shouldReadFromVar(property, propValue, fieldValue)) 
+                {
+                    adapter.setProperty(cfg, category, f, fieldValue);
+                } 
+                else
+                {
+                    set(instance, f, propValue);
+                }
+                return;
+            }
+            else
+            {
+                prop = adapter.getProp(cfg, category, f, instance, comment);
+                set(instance, f, adapter.getValue(prop));
+            }
```

Comment:
Also, cant this all be simplified down to the same prop = cfg.getCat().getProp(name), remove the returns, and wrap the code at the end of the function in a if (loading) setting the comment? because the comment is really the only difference between the two sides of this if statement.